ci/dockerfile/ubuntu24.04.dockerfile

```diff
@@ -0,0 +1,123 @@
+FROM ubuntu:24.04
+
+ARG IS_VM
+
+# Add steps here to set up dependencies
+RUN apt-get update -y && env DEBIAN_FRONTEND=noninteractive apt-get install -y \
+    autoconf \
+    bc \
+    bison \
+    build-essential \
+    cargo \
+    clang \
+    curl \
+    flex \
+    gawk \
+    file \
+    gdb \
+    gettext \
+    git \
+    gnupg \
+    golang \
+    jq \
+    libapr1-dev \
+    libaprutil1-dev \
+    libelf-dev \
+    libevent-dev \
+    libexpat1 \
+    libexpat1-dev \
+    libjpeg-dev \
+    libmemcached-tools \
+    libnss-mdns \
+    libnss-myhostname \
+    libnuma1 \
+    libomp-dev \
+    libpcre2-dev \
+    libpcre3-dev \
+    libprotobuf-c-dev \
+    libssl-dev \
+    libunwind8 \
+    libxfixes3 \
+    libxi6 \
+    libxml2-dev \
+    libxrender1 \
+    libxxf86vm1 \
+    libjudy-dev \
+    libpixman-1-dev \
+    libipsec-mb-dev \
+    lsb-release \
+    lsof \
+    musl \
+    musl-tools \ 
+    nasm \
+    net-tools \
+    netcat-openbsd \
+    nginx \
+    ninja-build \
+    nodejs \
+    openjdk-11-jdk \
+    pkg-config \
+    protobuf-c-compiler \
+    protobuf-compiler \
+    pylint \
+    python3-apport \
+    python3-apt \
+    python3-breathe \
+    python3-click \
+    python3-cryptography \
+    python3-jinja2 \
+    python3-lxml \
+    python3-numpy \
+    python3-pandas \
+    python3-pil \
+    python3-pip \
+    python3-pkg-resources \
+    python3-protobuf \
+    python3-pyelftools \
+    python3-pytest \
+    python3-pytest-xdist \
+    python3-recommonmark \
+    python3-scipy \
+    python3-sphinx-rtd-theme \
+    python3-voluptuous \
+    r-base \
+    shellcheck \
+    sphinx-doc \
+    stress-ng \
+    sqlite3 \
+    texinfo \
+    uthash-dev \
+    vim \
+    wget \
+    zlib1g \
+    zlib1g-dev \
+    gnupg2 \
```

Comment:
Libraries list should be sorted alphabetically, gnupg2, binutils and sudo should go above

Reply:
Modified